Deprecation warning: <sys/sysctl.h> header is deprecated and will be removed

Mainly FYI, building from HEAD (https://github.com/simsong/bulk_extractor/commit/bc5caa71e04906aeba387caff43c16017646d66f) gives me the following deprecation warning

```
In file included from main.cpp:53:
/usr/include/sys/sysctl.h:21:2: warning: #warning "The <sys/sysctl.h> header is deprecated and will be removed." [-Wcpp]
   21 | #warning "The <sys/sysctl.h> header is deprecated and will be removed."
      | 
```

--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,7 @@
+2020-06-14  fedora Cloud User  <fedora@ip-172-30-4-244.ec2.internal>
+
+	* src/threadpool.cpp: numCPU() now simply calls std::thread::hardware_concurrency(). Perhaps more should be replaced over time, but it would be super-exciting to see the threadpool be able to spin of threads onto other servers, which was the original intent. 
+
 2020-06-13  Simson Garfinkel <simsong@acm.org>
 	* updated license to MIT License, copyright Simson L. Garfinkel, consistent with the fact that this is no longer an official US Government work product.
 

--- a/configure.ac
+++ b/configure.ac
@@ -243,7 +243,7 @@ AC_SYS_POSIX_TERMIOS
 
 ## Check for headers used by bulk Extractor
 ## do not put pthread here
-AC_CHECK_HEADERS([alloca.h dirent.h dlfcn.h err.h errno.h fcntl.h inttypes.h libgen.h limits.h malloc.h mmap.h pwd.h signal.h stdarg.h stdint.h stdio.h strings.h string.h stdlib.h sys/cdefs.h sys/disk.h sys/fcntl.h sys/ioctl.h sys/mman.h sys/mmap.h sys/mount.h sys/param.h sys/socket.h sys/stat.h sys/types.h sys/time.h sys/resource.h sys/sysctl.h time.h unistd.h windows.h CoreServices/CoreServices.h])
+AC_CHECK_HEADERS([alloca.h dirent.h dlfcn.h err.h errno.h fcntl.h inttypes.h libgen.h limits.h malloc.h mmap.h pwd.h signal.h stdarg.h stdint.h stdio.h strings.h string.h stdlib.h sys/cdefs.h sys/disk.h sys/fcntl.h sys/ioctl.h sys/mman.h sys/mmap.h sys/mount.h sys/param.h sys/socket.h sys/stat.h sys/types.h sys/time.h sys/resource.h time.h unistd.h windows.h CoreServices/CoreServices.h])
 AC_CHECK_FUNCS([err errx getuid getpwuid gethostname getrusage gmtime_r isxdigit ishexnumber le64toh localtime_r _lseeki64 inet_ntop pread64 pread printf mmap munmap MD5 mkstemp mktemp random srandom srandomdev sleep SleepEx strptime usleep vasprintf warn warnx])
 AC_CHECK_FUNCS([CreateProcess LoadLibrary IncrementAtomic InterlockedIncrement])
 

--- a/etc/CONFIGURE_FEDORA32.bash
+++ b/etc/CONFIGURE_FEDORA32.bash
@@ -1,0 +1,100 @@
+#!/bin/bash
+LIBEWF_URL=https://github.com/libyal/libewf/releases/download/20171104/libewf-experimental-20171104.tar.gz
+FEDORA_VERSION=32
+cat <<EOF
+*******************************************************************
+        Configuring Amazon Linux for compiling bulk_extractor
+*******************************************************************
+
+This script will configure a fresh Amazon Linux system to compile 
+bulk_extractor.  Please perform the following steps:
+
+1. Start a VM
+2. sudo yum -y update
+3. sudo yum -y install git
+4. git clone https://github.com/simsong/bulk_extractor.git
+5. cd bulk_extractor
+6. sudo sh misc/CONFIGURE_AMAZON_LINUX.bash
+7. make && sudo make install
+
+press any key to continue...
+EOF
+read
+
+# cd to the directory where the script is
+# http://stackoverflow.com/questions/59895/can-a-bash-script-tell-what-directory-its-stored-in
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+if [ "$PWD" != "$DIR" ]; then
+    changed_dir="true"
+else
+    changed_dir="false"
+fi
+cd $DIR
+
+MPKGS="autoconf automake make flex gcc gcc-c++ git libtool "
+MPKGS+="md5deep wget bison zlib-devel "
+MPKGS+="libewf libewf-devel java-1.8.0-openjdk-devel "
+MPKGS+="libxml2-devel libxml2-static openssl-devel "
+MPKGS+="sqlite-devel expat-devel "
+
+if [ ! -r /etc/os-release ]; then
+  echo This requires Fedora Linux
+  exit 1
+fi
+
+. /etc/os-release
+if [ $ID != 'fedora' ]; then
+    echo This requires Fedora Linux. You have $ID.
+    exit 1
+fi
+
+if [ $VERSION_ID -ne $FEDORA_VERSION ]; then
+    echo This requires version $FEDORA_VERSION of $ID. You have $VERSION_ID. 
+    exit 1
+fi
+
+echo Will now try to install 
+
+sudo yum install -y $MPKGS --skip-broken
+if [ $? != 0 ]; then
+  echo "Could not install some of the packages. Will not proceed."
+  exit 1
+fi
+
+echo
+echo "Now performing a yum update to update system packages"
+sudo yum -y update
+
+
+LIBEWF_FNAME=`echo $LIBEWF_URL| sed s:.*/::`
+LIBEWF_DIR=`echo $LIBEWF_FNAME | sed s/-experimental// | sed s/.tar.gz//`
+echo 
+echo "Now installing libewf"
+wget https://github.com/libyal/libewf/releases/download/20171104/libewf-experimental-20171104.tar.gz
+tar xfvz $LIBEWF_FNAME
+pushd $LIBEWF_DIR
+./configure && make
+sudo make install
+popd
+
+# Make sure that /usr/local/lib is in ldconfig
+sudo /bin/rm -f /tmp/local.conf
+echo /usr/local/lib > /tmp/local.conf
+sudo mv /tmp/local.conf /etc/ld.so.conf.d/local.conf
+sudo ldconfig
+
+#
+#
+#
+
+echo ...
+echo 'Now running ../bootstrap.sh and configure'
+pushd ..
+sh bootstrap.sh
+sh configure
+popd
+echo ================================================================
+echo ================================================================
+echo 'You are now ready to compile bulk_extractor for Linux.'
+echo 'To compile, type make'
+echo 'To make a distribution, type make release'

--- a/src/main.cpp
+++ b/src/main.cpp
@@ -49,10 +49,6 @@
 #include <sys/resource.h>
 #endif
 
-#ifdef HAVE_SYS_SYSCTL_H
-#include <sys/sysctl.h>
-#endif
-
 #ifdef WIN32 
 // Allows us to open standard input in binary mode by default 
 // See http://gnuwin32.sourceforge.net/compile.html for more 

--- a/src/scan_net.cpp
+++ b/src/scan_net.cpp
@@ -281,7 +281,7 @@ static uint16_t IPv6L3Chksum(const sbuf_t &sbuf, u_int chksum_byteoffset)
 static uint16_t cksum(const struct be13::ip4 * const ip, int len) 
 {
     long  sum = 0;  /* assume 32 bit long, 16 bit short */
-    const uint16_t *ipp = (uint16_t *) ip;
+    const uint16_t *ipp = (const uint16_t *) ip;
     int   octets_processed = 0;
 
     while (len > 1) {

--- a/src/threadpool.cpp
+++ b/src/threadpool.cpp
@@ -7,51 +7,23 @@
 #include <dirent.h>
 #include <ctype.h>
 #include <fcntl.h>
-#include <set>
+#include <unistd.h>
 #include <setjmp.h>
+
+#include <set>
 #include <vector>
 #include <queue>
-#include <unistd.h>
+#include <thread>
 
 
 /* Return the number of CPUs we have on various architectures.
  * From http://stackoverflow.com/questions/150355/programmatically-find-the-number-of-cores-on-a-machine
+ * 2020-06-14 - slg - Migrated to std::thread::hardware_concurrency() per C++11.
  */
 
 u_int threadpool::numCPU()
 {
-    int numCPU=1;			// default
-#ifdef WIN32
-    SYSTEM_INFO sysinfo;
-    GetSystemInfo( &sysinfo );
-    numCPU = sysinfo.dwNumberOfProcessors;
-#endif
-#if defined(HW_AVAILCPU) && defined(HW_NCPU)
-    int mib[4];
-    size_t len=sizeof(numCPU);
-
-    /* set the mib for hw.ncpu */
-    memset(mib,0,sizeof(mib));
-    mib[0] = CTL_HW;
-    mib[1] = HW_AVAILCPU;  // alternatively, try HW_NCPU;
-
-    /* get the number of CPUs from the system */
-    if(sysctl(mib, 2, &numCPU, &len, NULL, 0)){
-	perror("sysctl");
-    }
-
-    if( numCPU <= 1 ) {
-	mib[1] = HW_NCPU;
-	sysctl( mib, 2, &numCPU, &len, NULL, 0 );
-	if( numCPU < 1 ) {
-	    numCPU = 1;
-	}
-    }
-#endif
-#ifdef _SC_NPROCESSORS_ONLN
-    numCPU = sysconf(_SC_NPROCESSORS_ONLN);
-#endif
-    return numCPU;
+    return std::thread::hardware_concurrency();
 }
 
 #ifdef WIN32

